### PR TITLE
Add Language Menu

### PR DESCRIPTION
**Pull request by sithlord48**
- Add a menuLanguage that is auto populated by the qm files found in the `appdir/i18n` directory.
  - Update language files to include the language name for use in this menu
  - Dynamic Language Switching
  - Added a few translated string to French and Spanish ts files.
  - English translation Finished.

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -70,12 +76,39 @@ void MainWindow::detectTranslations()
...
+    }
+    action->setChecked(true);
+    QTranslator translator;
+    QString title,message;
+    translator.load(QString("FrogArchiver_%1.qm").arg(m_settings->value("language").toString()), QString("%1/i18n").arg(QCoreApplication::applicationDirPath()));
+    title= translator.translate("MainWindow","Language Changed");
+    if(title.isEmpty()){title = "Language Changed";}
+    message = translator.translate("MainWindow","You must restart for the language to change.");
+    if(message.isEmpty()){message = "You must restart for the language to change.";}
+    QMessageBox::information(this,title,message);
```
Comment (berviantoleo) [inline src/mainwindow.cpp]: I am curious, is it possible just auto restart the app after the language changes?

Comment (berviantoleo) [reply src/mainwindow.cpp]: Or at least we give a prompt dialog and the "OK" button will restart the app.

Comment (sithlord48) [reply src/mainwindow.cpp]: Not sure about restart we can have the application close easy enough

Comment (berviantoleo) [reply src/mainwindow.cpp]: I found this: https://wiki.qt.io/How_to_make_an_Application_restartable

What do you think?

Comment (sithlord48) [reply src/mainwindow.cpp]: Not a Fan .. Restarts are hacky..

Comment (berviantoleo) [reply src/mainwindow.cpp]: Okay. :smile: If you think this PR is ready. I'll merge this PR. Thanks.

**Review by berviantoleo — APPROVED** [review]
lgtm

**Pull request merged**